circuits/types/field/src/compare.rs

```diff
@@ -21,25 +21,52 @@ impl<E: Environment> Compare<Field<E>> for Field<E> {
 
     /// Returns `true` if `self` is less than `other`.
     fn is_less_than(&self, other: &Self) -> Self::Boolean {
-        let mut is_less_than = Boolean::constant(false);
-        let mut are_previous_bits_equal = Boolean::constant(true);
-
-        // Initialize an iterator over `self` and `other` from MSB to LSB.
-        let self_bits_be = self.to_bits_be();
-        let other_bits_be = other.to_bits_be();
-        let bits_be = self_bits_be.iter().zip_eq(&other_bits_be);
-
-        for (index, (self_bit, other_bit)) in bits_be.enumerate() {
-            // Determine if `self` is less than `other` up to the `index`-th bit.
-            is_less_than |= &are_previous_bits_equal & (!self_bit & other_bit);
-
-            // Skip the update to the LSB, as this boolean is subsequently discarded.
-            if index != self_bits_be.len() - 1 {
-                are_previous_bits_equal &= self_bit.is_equal(other_bit);
-            }
+        if self.is_constant() && other.is_constant() {
+            Boolean::constant(self.eject_value() < other.eject_value())
+        } else if self.is_constant() {
+            // (For advanced users) this implementation saves us from instantiating 253 constants for
+            // the bits of `self`. The implementation in the `else` case invokes `to_bits_le` on
+            // `self` which would allocate 253 constants. Since `self` is constant, we can directly
+            // inspect its bits and construct an equivalent but reduced ternary expression.
+            self.eject_value().to_bits_le().into_iter().zip_eq(other.to_bits_le()).fold(
+                Boolean::constant(false),
+                |rest_is_less, (self_bit, other_bit)| {
+                    if self_bit {
+                        Boolean::ternary(&!&other_bit, &other_bit, &rest_is_less)
```

Comment:
This expression is a bit difficult to read; might be worth explaining what is going on here.